cognite/client/data_classes/_base.py

```diff
@@ -281,14 +285,32 @@ def get(self, id: Optional[int] = None, external_id: Optional[str] = None) -> Op
             return self._id_to_item.get(id)
         return self._external_id_to_item.get(external_id)
 
-    def to_pandas(self, camel_case: bool = False) -> pandas.DataFrame:
+    def to_pandas(
+        self,
+        camel_case: bool = False,
+        expand_metadata: bool = False,
+        metadata_prefix: str = "metadata.",
+    ) -> pandas.DataFrame:
         """Convert the instance into a pandas DataFrame.
 
+        Args:
+            camel_case (bool): Convert column names to camel case (e.g. `externalId` instead of `external_id`)
+            expand_metadata (bool): Expand the metadata column into separate columns.
+            metadata_prefix (str): Prefix to use for metadata columns.
+
         Returns:
             pandas.DataFrame: The dataframe.
         """
         pd = cast(Any, utils._auxiliary.local_import("pandas"))
         df = pd.DataFrame(self.dump(camel_case=camel_case))
+
+        if expand_metadata and "metadata" in df.columns:
+            meta_df = pd.json_normalize(df["metadata"])
+            meta_df_prefix_cols = [f"{metadata_prefix}{col}" for col in meta_df.columns]
+            if common_cols := set(meta_df_prefix_cols).intersection(set(df.columns)):
+                raise ValueError(f"Metadata contains columns that are already present in the dataframe: {common_cols}")
+            df = pd.concat([df.drop("metadata", axis=1), meta_df.add_prefix(metadata_prefix)], axis=1)
```

Comment:
Some 🏌
```suggestion
            meta_df = pd.json_normalize(df["metadata"]).add_prefix(metadata_prefix)
            if common_cols := set(meta_df.columns).intersection(set(df.columns)):
                raise ValueError(f"Metadata contains columns that are already present in the dataframe: {common_cols}")
            df = pd.concat([df.drop("metadata", axis=1), meta_df], axis=1)
```

Reply:
Hang on, don't we want to leave the choice in the hands of the user @doctrino? Should we not keep the `if expand_metadata` bit in? Otherwise this might be a breaking change, right?

Reply:
Just a bad copy from me, I lost the first line. Fixed it now.

Reply:
the main argument for doing this change is to avoid doing the same operation (adding a prefix) in two different ways. If there were a bug in `[f"{metadata_prefix}{col}" for col in meta_df.columns]` it could pass the ValueError check, even though it should not have.

Reply:
Yes, good point. 👍